distros/catkin-nixpkgs-prefix-path.patch

```diff
@@ -0,0 +1,10 @@
+--- a/cmake/all.cmake
++++ b/cmake/all.cmake
+@@ -41,6 +41,7 @@
+ 
+ # use either CMAKE_PREFIX_PATH explicitly passed to CMake as a command line argument
+ # or CMAKE_PREFIX_PATH from the environment
++string(REPLACE ":" ";" CMAKE_PREFIX_PATH $ENV{NIXPKGS_CMAKE_PREFIX_PATH})
```

Comment:
Wrap it in `if(NOT DEFINED CMAKE_PREFIX_PATH)` so that the user can always override it, if needed.